internal/jet/func_expression.go

```diff
@@ -843,3 +855,42 @@ func newTimestampzFunc(name string, expressions ...Expression) *timestampzFunc {
 func Func(name string, expressions ...Expression) Expression {
 	return NewFunc(name, expressions, nil)
 }
+
+type rangeFunc[T Expression] struct {
```

Comment:
I don't think `rangeFunc[T Expression]` is needed, because you can do always do `RangeExp[T](NewFunc(....))`